packages/kbn-reporting/generate_csv/types/index.d.ts

```diff
@@ -0,0 +1,274 @@
+export const UI_SETTINGS_CSV_SEPARATOR = 'csv:separator';
+export const UI_SETTINGS_CSV_QUOTE_VALUES = 'csv:quoteValues';
+export const UI_SETTINGS_DATEFORMAT_TZ = 'dateFormat:tz';
+
+type ReportTypeDeclaration = typeof reportTypes;
+export type ReportTypes = ReportTypeDeclaration[keyof ReportTypeDeclaration];
+
+type JobTypeDeclaration = typeof jobTypes;
+export type JobTypes = JobTypeDeclaration[keyof JobTypeDeclaration];
+
+export const CSV_SEARCHSOURCE_IMMEDIATE_TYPE = 'csv_searchsource_immediate';
+
+// This is deprecated because it lacks support for runtime fields
+// but the extension points are still needed for pre-existing scripted automation, until 8.0
+export const CSV_REPORT_TYPE_DEPRECATED = 'CSV';
+export const CSV_JOB_TYPE_DEPRECATED = 'csv';
+
+export const USES_HEADLESS_JOB_TYPES = [
+  PDF_JOB_TYPE,
+  PNG_JOB_TYPE,
+  PDF_JOB_TYPE_V2,
+  PNG_JOB_TYPE_V2,
+];
+
+export const DEPRECATED_JOB_TYPES = [CSV_JOB_TYPE_DEPRECATED];
+
+// Licenses
+export const LICENSE_TYPE_TRIAL = 'trial';
+export const LICENSE_TYPE_BASIC = 'basic';
+export const LICENSE_TYPE_CLOUD_STANDARD = 'standard';
+export const LICENSE_TYPE_GOLD = 'gold';
+export const LICENSE_TYPE_PLATINUM = 'platinum';
+export const LICENSE_TYPE_ENTERPRISE = 'enterprise';
+
+// Routes
+export const API_BASE_URL = '/api/reporting'; // "Generation URL" from share menu
+export const API_BASE_GENERATE = `${API_BASE_URL}/generate`;
+export const API_LIST_URL = `${API_BASE_URL}/jobs`;
+export const API_DIAGNOSE_URL = `${API_BASE_URL}/diagnose`;
+
+export const API_GET_ILM_POLICY_STATUS = `${API_BASE_URL}/ilm_policy_status`;
+export const API_MIGRATE_ILM_POLICY_URL = `${API_BASE_URL}/deprecations/migrate_ilm_policy`;
+export const API_BASE_URL_V1 = '/api/reporting/v1'; //
+
+export const ILM_POLICY_NAME = 'kibana-reporting';
+
+// Usage counter types
+export const API_USAGE_COUNTER_TYPE = 'reportingApi';
+export const API_USAGE_ERROR_TYPE = 'reportingApiError';
+
+// Management UI route
+export const REPORTING_MANAGEMENT_HOME = '/app/management/insightsAndAlerting/reporting';
+
+export const REPORTING_REDIRECT_LOCATOR_STORE_KEY = '__REPORTING_REDIRECT_LOCATOR_STORE_KEY__';
+
+// Export Reporting Type Definitions
+export const CSV_REPORT_TYPE = 'CSV';
+export const CSV_REPORT_TYPE_V2 = 'csv_v2';
+
+export const PDF_REPORT_TYPE = 'printablePdf';
+export const PDF_REPORT_TYPE_V2 = 'printablePdfV2';
+
+export const PNG_REPORT_TYPE = 'PNG';
+export const PNG_REPORT_TYPE_V2 = 'pngV2';
+
+// Export Job Type Definitions
+export const CSV_JOB_TYPE = 'csv_searchsource';
+
+export const PDF_JOB_TYPE = 'printable_pdf';
+export const PDF_JOB_TYPE_V2 = 'printable_pdf_v2';
+
+export const PNG_JOB_TYPE = 'PNG';
+export const PNG_JOB_TYPE_V2 = 'PNGV2';
+
+/**
+ * A way to get the client side route for the reporting redirect app.
+ *
+ * TODO: Add a job ID and a locator to use so that we can redirect without expecting state to
+ * be injected to the page
+ */
+export const getRedirectAppPath = () => {
```

Comment:
I think there's a lot of things in here that aren't related to CSV generation.. I hope not all of this is, anyway :)